skiko/src/commonTest/kotlin/org/jetbrains/skia/util/assertCloseEnough.kt

```diff
@@ -21,10 +23,39 @@ internal fun assertCloseEnough(expected: Matrix33, actual: Matrix33?, epsilon: F
     expected.mat.zip(actual!!.mat).forEach { (a, b) -> assertCloseEnough(a, b, epsilon) }
 }
 
-
 internal fun assertCloseEnough(expected: Color4f, actual: Color4f?, epsilon: Float = EPSILON) {
     assertCloseEnough(expected.r, actual!!.r, epsilon)
     assertCloseEnough(expected.g, actual.g, epsilon)
     assertCloseEnough(expected.b, actual.b, epsilon)
     assertCloseEnough(expected.a, actual.a, epsilon)
-}
\ No newline at end of file
+}
+
+internal fun assertCloseEnough(expected: Rect, actual: Rect?, epsilon: Float = EPSILON) {
+    assertCloseEnough(expected.left, actual!!.left, epsilon)
+    assertCloseEnough(expected.top, actual.top, epsilon)
+    assertCloseEnough(expected.right, actual.right, epsilon)
+    assertCloseEnough(expected.bottom, actual.bottom, epsilon)
+}
+
+internal fun assertContentCloseEnough(expected: FloatArray, actual: FloatArray?, epsilon: Float = EPSILON) {
+    if (actual == null) {
+        throw AssertionError("expected $expected, got null")
+    }
+
+    for (i in expected.indices) {
+        if (abs(expected[i] - actual[i]) > epsilon) {
+            throw AssertionError("results differ at index$i, expected ${expected[i]}, got ${actual[i]}")
+        }
+    }
+}
+
+internal fun assertContentCloseEnough(expected: Array<Point>, actual: Array<Point>?, epsilon: Float = EPSILON) {
+    assertNotNull(actual)
```

Comment:
a similar question about whether we need an assertion - I mean we still fail with exception if we'll try t cast.